language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java

```diff
@@ -73,7 +73,7 @@ public static List<LSCompletionItem> getRecordFieldCompletionItems(BallerinaComp
             CompletionItem fieldItem = new CompletionItem();
             fieldItem.setInsertText(insertText);
             fieldItem.setInsertTextFormat(InsertTextFormat.Snippet);
-            fieldItem.setLabel(name);
+            fieldItem.setLabel(CommonUtil.escapeReservedKeyword(name));
```

Comment:
From where are you getting this `name` var from? If it's from the symbol for the field, then there shouldn't be any need for you to manually escape it like this. When getting the name from the symbol it should give the correctly escaped name :thinking:

Reply:
We use `RecordTypeSymbol.fieldDescriptors()` map to determine the field names and their symbols. We use the keys of that map as names.